modules/storages/app/models/storages/storage.rb

```diff
@@ -82,6 +80,10 @@ class Storages::Storage < ApplicationRecord
     end
   end
 
+  scope :configured, -> do
+    subclasses.map { |clazz| clazz.configured.to_a }.flatten
```

Comment:
this is very much to be discussed. Any better idea I'm fine to try. :D

IMHO, we need a scope on the base class, that can be executed e.g. in the API or the project_storages controller, like `Storages::Storage.configured`. To make that work, I currently execute the scopes of all subclasses and concatenate the results. This is working, but it might appear a bit smelly.

WDYT?

Reply:
I'd suggest to not implement the scope if we do not need it explicitly somewhere. Maybe, let's use `raise NotImplementedError` here as well.

Reply:
I'm confused? Like I said, we DO need the scope.

In the locations, where we use it, you do not want to fetch first all provider types, to iterate over them and call the scope on each. you do want to say: "Give me all configured storages, please!"

Reply:
discussion cancellation: I removed the scope and replaced the usage with the instance method `configured?`

Kudos to @mereghost for brainstorming.